docs/source/tutorial-rb.rst

```diff
@@ -0,0 +1,789 @@
+of the |parameters| of `m`:
+
+.. jupyter-execute::
+
+    parameter_space = fom.parameters.space(0.0001, 1.)
+
+Here, `0.0001` and `1.` are the common lower and upper bounds for the
+individual components of all |parameters| of `fom`. In our case `fom`
+expects a single parameter `'diffusion'` of 9 values:
+
+.. jupyter-execute::
+
+   fom.parameters
+
+If `fom` were to depend on multiple |parameters|, we could also call the
+:meth:`~pymor.parameters.base.Parameters.space` method with a dictionary
+of lower and upper bounds per parameter.
+
+The main use of |ParameterSpaces| in pyMOR is that they allow to easily sample
+|parameter values| from their domain using the methods
+:meth:`~pymor.parameters.base.ParameterSpace.sample_uniformly` and
+:meth:`~pymor.parameters.base.ParameterSpace.sample_randomly`.
+
+
+Computing the snapshot data
+---------------------------
+
+Reduced basis methods are snapshot-based, which means that they build
+the reduced space as a linear subspace of the linear span of solutions
+of the `fom` for certain |parameter values|. The easiest
+approach is to just pick these values randomly, what we will do in the
+following. First we define a training set of 25 parameters:
+
+.. jupyter-execute::
+
+    training_set = parameter_space.sample_randomly(25)
+    print(training_set)
+
+Then we :meth:`~pymor.models.interface.Model.solve` the `fom`
+for all |parameter values| in the training set and accumulate all
+solution vectors in a single |VectorArray| using its
+:meth:`~pymor.vectorarrays.interface.VectorArray.append` method. But first
+we need to create an empty |VectorArray| to which the solutions can
+be appended. New |VectorArrays| in pyMOR are always created by a
+corresponding |VectorSpace|. Empty arrays are created using the
+:meth:`~pymor.vectorarrays.interface.VectorSpace.empty` method. But what
+is the right |VectorSpace|? All :meth:`solutions <pymor.models.interface.Model.solve>`
+of a |Model| belong to its :attr:`~pymor.models.interface.Model.solution_space`,
+so we write:
+
+.. jupyter-execute::
+
+    U = fom.solution_space.empty()
+    for mu in training_set:
+        U.append(fom.solve(mu))
+
+Note that `fom.solve` returns a |VectorArray| containing a single vector.
+This entire array (of one vector) is then appended to the `U` array.
+pyMOR has no notion of single vectors, we only speak of |VectorArrays|.
+
+What exactly is a |VectorSpace|? A |VectorSpace| in pyMOR holds all
+information necessary to build |VectorArrays| containing vector objects
+of a certain type. In our case we have
+
+.. jupyter-execute::
+
+   fom.solution_space
+
+which means that the created |VectorArrays| will internally hold
+|NumPy arrays| for data storage. The number is the dimension of the
+vector. We have here a |NumpyVectorSpace| because pyMOR's builtin
+discretizations are built around the NumPy/SciPy stack. If `fom` would
+represent a |Model| living in an external PDE solver, we would have
```

Comment:
Maybe reference to the tutorial for external PDE solvers?

Reply:
Can't do that yet ..

Reply:
You're right, maybe we could keep that in mind and add a reference later on.